patching/src/main/java/org/jboss/as/patching/runner/PatchModuleInvalidationUtils.java

```diff
@@ -189,7 +189,7 @@ private static void updateJar(final File file, final byte[] searchPattern, final
                 if (!validateEndRecord(file, channel, pos, endSig)) {
                     pos = scanForEndSig(file, channel, context);
                 }
-                if (pos == -1) {
+                if (pos == -1 && context != null) {
```

Comment:
The scanForEndSig call would fail if context were null.

I think the thing to do with this class is change L186 to throw an IllegalArgumentException. The purpose of this method is to write either CRIPPLED_ENDSIG or GOOD_ENDSIG; if neither is passed in it should fail.

Reply:
Better is to replace the newSig param with a boolean and use CRIPPLED_ENDSIG or GOOD_ENDSIG based on that.

Reply:
@bstansberry Hello Brian, thanks for your recommendation. It's fixed. Could you review/merge, please?